packages/core-data/src/entity-provider.js

```diff
@@ -171,23 +171,15 @@ export function useEntityBlockEditor( kind, name, { id: _id } = {} ) {
 	const { __unstableCreateUndoLevel, editEntityRecord } =
 		useDispatch( STORE_NAME );
 
-	useEffect( () => {
-		// Load the blocks from the content if not already in state
-		// Guard against other instances that might have
-		// set content to a function already or the blocks are already in state.
-		if ( content && typeof content !== 'function' && ! blocks ) {
-			const parsedContent = parse( content );
-			editEntityRecord(
-				kind,
-				name,
-				id,
-				{
-					blocks: parsedContent,
-				},
-				{ undoIgnore: true }
-			);
+	const blocks = useMemo( () => {
+		if ( editedBlocks ) {
+			return editedBlocks;
 		}
-	}, [ content ] );
+
+		return content && typeof content !== 'function'
+			? parse( content )
+			: EMPTY_ARRAY;
+	}, [ editedBlocks, content ] );
```

Comment:
> Setting blocks to undefined means that on undo, they will need to be parsed from scratch from the content string and all client IDs will be different.
> This will cause all blocks' edit components to be unmounted and re-created from scratch, on a simple undo operation that shouldn't do this. It should just update one little attribute of one little block.

I agree that ideally the components should be unmounted and recreated from scratch but making the "blocks" edit `undefined` on undo is (in isolation) a fine.

That said, I think maybe we need a way to tell `core-data` to load these transient values on initial "fetch" of the entities instead of leaving that to consumer code which will always lead to inconsistencies.

Reply:
> making the "blocks" edit `undefined` on undo is (in isolation) a fine.

In this case it's not fine, because conceptually the `blocks` value is known and it's not `undefined`. It just lives outside the data store, in React state. Losing that value leads to losing information, namely the blocks' already assigned client IDs.

I don't know how to fix it, and the issue is not terribly urgent. I mainly wanted to document that it exists and write down what I found.